tools/lint.sh

```
@@ -212,12 +212,402 @@ fi
+            done <<< "$VOLATILE_VAR_NAMES"
+        fi
+    done
+fi
+
+ISSUES_FOUND=false
+
+if [ -n "$VOLATILE_ISSUES" ]; then
+    echo -e "${YELLOW}⚠ Potential missing 'volatile' on IRQ-shared variables:${NC}"
+    echo ""
+    echo -e "$VOLATILE_ISSUES" | head -10
+    [ $(echo -e "$VOLATILE_ISSUES" | wc -l) -gt 10 ] && echo "... and more"
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)